exotica_python/src/Exotica.cpp

```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
-    motionSolver.def(
-        "solve", [](std::shared_ptr<MotionSolver> sol) { return Solve(sol); },
+    py::class_<TaskMap, std::shared_ptr<TaskMap>, Object> task_map(module, "TaskMap");
+    task_map.def_readonly("id", &TaskMap::Id);
+    task_map.def_readonly("start", &TaskMap::Start);
+    task_map.def_readonly("length", &TaskMap::Length);
+    task_map.def_readonly("startJ", &TaskMap::StartJ);
+    task_map.def_readonly("lengthJ", &TaskMap::LengthJ);
+    task_map.def("task_space_dim", (int (TaskMap::*)()) & TaskMap::taskSpaceDim);
+    task_map.def("task_Space_jacobian_dim", &TaskMap::taskSpaceJacobianDim);
+
+    py::class_<TimeIndexedTask, std::shared_ptr<TimeIndexedTask>> time_indexed_task(module, "TimeIndexedTask");
+    time_indexed_task.def_readonly("PhiN", &TimeIndexedTask::PhiN);
+    time_indexed_task.def_readonly("JN", &TimeIndexedTask::JN);
+    time_indexed_task.def_readonly("num_tasks", &TimeIndexedTask::NumTasks);
+    time_indexed_task.def_readonly("y", &TimeIndexedTask::y);
+    time_indexed_task.def_readonly("ydiff", &TimeIndexedTask::ydiff);
+    time_indexed_task.def_readonly("Phi", &TimeIndexedTask::Phi);
+    // time_indexed_task.def_readonly("H", &TimeIndexedTask::H);
+    time_indexed_task.def_readonly("J", &TimeIndexedTask::J);
+    time_indexed_task.def_readonly("S", &TimeIndexedTask::S);
+    time_indexed_task.def_readonly("T", &TimeIndexedTask::T);
+    time_indexed_task.def_readonly("Tasks", &TimeIndexedTask::Tasks);
+    time_indexed_task.def_readonly("TaskMaps", &TimeIndexedTask::TaskMaps);
+
+    py::class_<EndPoseTask, std::shared_ptr<EndPoseTask>> end_pose_task(module, "EndPoseTask");
+    end_pose_task.def_readonly("PhiN", &EndPoseTask::PhiN);
+    end_pose_task.def_readonly("JN", &EndPoseTask::JN);
+    end_pose_task.def_readonly("num_tasks", &EndPoseTask::NumTasks);
+    end_pose_task.def_readonly("y", &EndPoseTask::y);
+    end_pose_task.def_readonly("ydiff", &EndPoseTask::ydiff);
+    end_pose_task.def_readonly("Phi", &EndPoseTask::Phi);
+    // end_pose_task.def_readonly("H", &EndPoseTask::H);
+    end_pose_task.def_readonly("J", &EndPoseTask::J);
+    end_pose_task.def_readonly("S", &EndPoseTask::S);
+    end_pose_task.def_readonly("Tasks", &EndPoseTask::Tasks);
+    end_pose_task.def_readonly("TaskMaps", &EndPoseTask::TaskMaps);
+
+    py::class_<SamplingTask, std::shared_ptr<SamplingTask>> sampling_task(module, "SamplingTask");
+    sampling_task.def_readonly("PhiN", &SamplingTask::PhiN);
+    sampling_task.def_readonly("JN", &SamplingTask::JN);
+    sampling_task.def_readonly("num_tasks", &SamplingTask::NumTasks);
+    sampling_task.def_readonly("y", &SamplingTask::y);
+    sampling_task.def_readonly("ydiff", &SamplingTask::ydiff);
+    sampling_task.def_readonly("Phi", &SamplingTask::Phi);
+    sampling_task.def_readonly("S", &SamplingTask::S);
+    sampling_task.def_readonly("Tasks", &SamplingTask::Tasks);
+    sampling_task.def_readonly("TaskMaps", &SamplingTask::TaskMaps);
```

Comment:
Use same API as `planning_problem`

Reply:
No, one is an attribute and one is a getter (method). So I will change capitalisation but we cannot make API equal.